src/rpc/handlers/LedgerEntry.cpp

```diff
@@ -214,7 +230,7 @@ tag_invoke(boost::json::value_from_tag, boost::json::value& jv, LedgerEntryHandl
         {JS(validated), output.validated},
         {JS(index), output.index},
     };
-
+    object["deleted_ledger_index"] = output.deleted_ledger_index.has_value()? output.deleted_ledger_index.value() : 0;
```

Comment:
you should use an if-statement to follow what the nodeBinary did. If the object has not been deleted, the response should omit the `deleted_ledger_index` field. Currently, your code would return `deleted_ledger_index: 0`